aws/resource_aws_glue_catalog_database_test.go

```diff
@@ -0,0 +1,128 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/glue"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSGlueCatalogDatabase_basic(t *testing.T) {
+	rInt := acctest.RandInt()
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGlueDatabaseDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccGlueCatalogDatabase_basic(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGlueCatalogDatabaseExists("aws_glue_catalog_database.test"),
+					resource.TestCheckResourceAttr(
+						"aws_glue_catalog_database.test",
+						"description",
+						"A test catalog from terraform",
+					),
+					resource.TestCheckResourceAttr(
+						"aws_glue_catalog_database.test",
+						"location_uri",
+						"my-location",
+					),
+					resource.TestCheckResourceAttr(
+						"aws_glue_catalog_database.test",
+						"parameters.param1",
+						"value1",
+					),
+					resource.TestCheckResourceAttr(
+						"aws_glue_catalog_database.test",
+						"parameters.param2",
+						"1",
+					),
+					resource.TestCheckResourceAttr(
+						"aws_glue_catalog_database.test",
+						"parameters.param3",
+						"50",
+					),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckGlueDatabaseDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).glueconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_glue_catalog_database" {
+			continue
+		}
+
+		input := &glue.GetDatabaseInput{
+			Name: aws.String(rs.Primary.ID),
+		}
+		if _, err := conn.GetDatabase(input); err != nil {
+			//Verify the error is what we want
+			if ae, ok := err.(awserr.Error); ok && ae.Code() == "EntityNotFoundException" {
```

Comment:
Two notes about this:
* We have a helper function which can let you remove the `awserr` import from this file
* I would recommend using the SDK constant here

`isAWSErr(err, glue.ErrCodeEntityNotFoundException, "")`